diff --git a/spectec/spec/6-reduction.watsup b/spectec/spec/6-reduction.watsup
index 3bfd50d2..34448752 100644
--- a/spectec/spec/6-reduction.watsup
+++ b/spectec/spec/6-reduction.watsup
@@ -320,7 +320,7 @@ rule Step/memory.grow-fail:
 
 rule Step_read/memory.fill-trap:
   z; (CONST I32 i) val (CONST I32 n) (MEMORY.FILL)  ~>  TRAP
-  -- if $(i + n) > |$mem(z, 0).DATA|
+  -- if $(i + n) > |$mem(z, x).DATA|
 
 rule Step_read/memory.fill-zero:
   z; (CONST I32 i) val (CONST I32 n) (MEMORY.FILL)  ~>  epsilon
@@ -359,7 +359,7 @@ rule Step_read/memory.copy-gt:
 
 rule Step_read/memory.init-trap:
   z; (CONST I32 j) (CONST I32 i) (CONST I32 n) (MEMORY.INIT x)  ~>  TRAP
-  -- if $(i + n) > |$data(z, x).DATA| \/ $(j + n) > |$mem(z, 0).DATA|
+  -- if $(i + n) > |$data(z, x).DATA| \/ $(j + n) > |$mem(z, x).DATA|
 
 rule Step_read/memory.init-zero:
   z; (CONST I32 j) (CONST I32 i) (CONST I32 n) (MEMORY.INIT x)  ~>  epsilon
