diff --git a/spectec/spec/6-reduction.watsup b/spectec/spec/6-reduction.watsup
index 3bfd50d2..0ffe9059 100644
--- a/spectec/spec/6-reduction.watsup
+++ b/spectec/spec/6-reduction.watsup
@@ -209,7 +209,7 @@ rule Step/table.grow-succeed:
   -- if $grow_table($table(z, x), n, ref) = ti
 
 rule Step/table.grow-fail:
-  z; ref (CONST I32 n) (TABLE.GROW x)  ~>  z; (CONST I32 $(-1))
+  z; (CONST I32 n) (TABLE.GROW x)  ~>  z; (CONST I32 $(-1))
 
 
 rule Step_read/table.fill-trap:
